src/Administration/Resources/app/administration/src/app/component/data-grid/sw-data-grid/index.js

```diff
@@ -374,20 +398,22 @@ Component.register('sw-data-grid', {
 
         onChangeCompactMode(value) {
             this.compact = value;
+            this.saveUserSettings();
         },
 
         onChangePreviews(value) {
             this.previews = value;
+            this.saveUserSettings();
         },
 
         onChangeColumnVisibility(value, index) {
             this.currentColumns[index].visible = value;
-            this.saveGridColumns();
```

Comment:
you cant remove the function call now... if a plugin has changed the saveGridColumns function, it wont work anymore.
Please find a way to call both methods and deprecate the function calls here.

Reply:
I re-added the call to `saveGridColumns` and added a deprecation marker - Is this fine for you the way I did it?

Reply:
yes, looks good to me.